lib/rexml/element.rb

```diff
@@ -2374,8 +2374,8 @@ def []=( name, value )
       end
 
       unless value.kind_of? Attribute
-        if @element.document and @element.document.doctype
-          value = Text::normalize( value, @element.document.doctype )
+        if @element.document and (doctype = @element.document.doctype)
+          value = Text::normalize( value, doctype )
```

Comment:
Is this really important?

Reply:
This line was responsible for `Enumerable#find` from the CPU profile. But I see it does not change much after this change, so reverting this change.